iam-login-service/src/test/java/it/infn/mw/iam/test/service/IamAccountServiceTests.java

```diff
@@ -843,4 +860,40 @@ public void testSetEndTimeWorksForNonNullDate() {
     assertThat(e.getAccount().getEndTime(), is(newEndTime));
   }
 
+  @Test
+  public void testNewAccountAddedToDefaultGroups() {
+    IamAccount account = cloneAccount(CICCIO_ACCOUNT);
+
+    IamGroup testGroup = new IamGroup();
+    testGroup.setName(TEST_GROUP_1);
+    Map<String, String> groupProperty = Map.of("name", TEST_GROUP_1, "enrollment", "INSERT");
+    List<Map<String, String>> defaultGroups = Arrays.asList(groupProperty);
+
+    
+    registrationProperties.setDefaultGroups(defaultGroups);   
+    when(iamGroupService.findByName(TEST_GROUP_1)).thenReturn(Optional.of(testGroup));
+
+    account = accountService.createAccount(account);
+
+    assertTrue(getGroup(account).equals(testGroup));
+  }
+
+  private IamGroup getGroup(IamAccount account) {
+    Optional<IamAccountGroupMembership> groupMembershipOptional = account.getGroups().stream().findFirst();
+    if(groupMembershipOptional.isPresent()){
+      return groupMembershipOptional.get().getGroup();      
+    }
+    return null;
+  }
+
+  @Test
+  public void testNoDefaultGroupsAddedWhenDefaultGroupsNotGiven() {
+    IamAccount account = cloneAccount(CICCIO_ACCOUNT);
+
+    account = accountService.createAccount(account);
+
+    Optional<IamAccountGroupMembership> groupMembershipOptional = account.getGroups().stream().findFirst();
+    assertFalse(groupMembershipOptional.isPresent());
+  }
+
```

Comment:
This is ok, it's a good mocking. Just for your knowledge (if needed) you can also write mvc tests and set properties by using a strategy like this https://github.com/indigo-iam/iam/blob/master/iam-login-service/src/test/java/it/infn/mw/iam/test/scim/me/ScimMeFullResponseEndpointTests.java#L44